std/traits.d

```diff
@@ -3340,6 +3341,183 @@ enum hasMember(T, string name) = __traits(hasMember, T, name);
     static assert(hasMember!(S, "foo"));
 }
 
+/++
+    Whether the symbol represented by the string, member, is a static member of
+    T.
+  +/
+template isStaticMember(T, string member)
+{
+    static if (!__traits(hasMember, T, member))
+        enum bool isStaticMember = false;
+    else
+    {
+        import std.meta : AliasSeq;
+        import std.traits : FunctionTypeOf;
+        alias sym = AliasSeq!(__traits(getMember, T, member))[0];
+
+        static if (__traits(isStaticFunction, sym))
+            enum bool isStaticMember = true;
+        else static if (is(FunctionTypeOf!sym == function) &&
```

Comment:
wrap the other checks inside a static if(is(sym == function)) to avoid templates for the most common case.